src/libraries/System.Private.CoreLib/src/System/SpanHelpers.T.cs

```diff
@@ -2631,19 +2618,19 @@ internal static int IndexOfAnyValueType<T>(ref T searchSpace, T value0, T value1
         }
 
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        private static int ComputeLastIndex<T>(ref T searchSpace, ref T current, Vector128<T> equals) where T : struct
+        private static int ComputeLastIndex<T>(nint offset, Vector128<T> equals) where T : struct
         {
             uint notEqualsElements = equals.ExtractMostSignificantBits();
             int index = 31 - BitOperations.LeadingZeroCount(notEqualsElements); // 31 = 32 (bits in Int32) - 1 (indexing from zero)
-            return (int)(Unsafe.ByteOffset(ref searchSpace, ref current) / Unsafe.SizeOf<T>()) + index;
+            return (int)(offset / Unsafe.SizeOf<T>()) + index;
```

Comment:
we don't need `SizeOf` anymore as `offset` is size-independent

```suggestion
            return (int)offset + index;
```